kaiax/valset/impl/api.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package impl
+
+import (
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/common"
+	"github.com/kaiachain/kaia/networks/rpc"
+)
+
+func (v *ValsetModule) APIs() []rpc.API {
+	return []rpc.API{
+		{
+			Namespace: "kaia",
+			Version:   "1.0",
+			Service:   newValidatorAPI(v),
+			Public:    true,
+		},
+		// TODO-kaiax-valset: add more namespaces. Istanbul? But it's not open to public. Anyway, move getDemoted to istanbul namespace.
+	}
+}
+
+type ValidatorAPI struct {
+	v *ValsetModule
+}
+
+// TODO-kaiax-valset: change the number parameter type to rpc.BlockNumberOrHash.
+func newValidatorAPI(v *ValsetModule) *ValidatorAPI {
+	return &ValidatorAPI{v}
+}
+
+// TODO-kaiax-valset: currently, this API returns the council list which is the result of block N-1. Is it correct?
+func (api *ValidatorAPI) GetCouncil(number *rpc.BlockNumber) ([]common.Address, error) {
+	header, err := headerByRpcNumber(api.v.chain, number)
+	if err != nil {
+		return nil, err
+	}
+	return api.v.GetCouncilAddressList(header.Number.Uint64() - 1)
```

Comment:
I thought `GetCouncil(N) = GetCouncilAddressList(N)`. Also the implementation of `GetCouncilAddressList` looks like we should pass `N`.